status-im/status-react
# Fix for versionCode conflicts when uploading to Play Store

### Summary:
This fixes how we manage build numbering. So far we simply used count of commits but that doesn't work when we cut custom release branches which vary in length. This can cause conflicts when uploading to Play Store, like in the following build:
https://jenkins.status.im/job/status-react/job/upload_release/21/
This results in error:
```
Google Api Error: apkUpgradeVersionConflict: APK specifies a version code that has already been used.
```
This is specifically caused by the `versionCode` app metadata set in `build.gradle`.

### Review notes (optional):
Currently the script tags the current commit with a new `build-${number}` tag but does not push.
This is left to the user/automation, which right now is done after builds for both platforms succeed.
Maybe a more correct approach would be to push right away, or only after we succeed in uploading, but I'm not sure.

### Testing notes (optional):
Try running it. It prints messages about what's happening to `stderr`.

### Steps to test:
Run `build_no.sh` with or without `--tag` and see what happens.

status: wip

<blockquote></blockquote>
